opensrp-chw-core/src/main/java/org/smartregister/chw/core/interactor/CoreChildProfileInteractor.java

```diff
@@ -465,6 +472,61 @@ public void fetchProfileTask(@NotNull Context context, @NotNull String baseEntit
         appExecutors.diskIO().execute(runnable);
     }
 
+    @Override
+    public void createCarePlanEvent(@NotNull Context context, @NotNull String encodedBundle, final CoreChildProfileContract.InteractorCallBack callback) {
+        Runnable runnable = () -> {
+            try {
+                // getting thinkMD id from encoded fhir bundle
+                String thinkMdId = ThinkMDLibrary.getInstance().getThinkMDPatientId(encodedBundle);
+                // getting the baseEntityId mapped to thinkMD
+                String baseEntityId = ChildDao.getBaseEntityID(context.getResources().getString(R.string.thinkmd_identifier_type), thinkMdId);
+                // creating the event to sync with server
+                if (baseEntityId != null) {
+                    Event carePlanEvent = ThinkMDLibrary.getInstance().createCarePlanEvent(encodedBundle,
+                            getFormTag(getAllSharedPreferences()),
+                            baseEntityId);
+
+                    JSONObject eventPartialJson = new JSONObject(JsonFormUtils.gson.toJson(carePlanEvent));
+                    ECSyncHelper.getInstance(context).addEvent(baseEntityId, eventPartialJson);
+
+                    appExecutors.mainThread().execute(callback::carePlanEventCreated);
+                }
+            } catch (Exception e) {
+                Timber.e(e);
+            }
+        };
+
+        appExecutors.diskIO().execute(runnable);
+    }
+
+    @Override
+    public void launchThinkMDHealthAssessment(@NotNull Context context) {
+
+    }
```

Comment:
This is not implemented

Reply:
This method is overridden and implemented in the CHW app.